posthog/hogql/grammar/HogQLLexer.g4

```diff
@@ -0,0 +1,307 @@
+FREEZE: F R E E Z E;
+FROM: F R O M;
+FULL: F U L L;
+FUNCTION: F U N C T I O N;
+GLOBAL: G L O B A L;
+GRANULARITY: G R A N U L A R I T Y;
+GROUP: G R O U P;
+HAVING: H A V I N G;
+HIERARCHICAL: H I E R A R C H I C A L;
+HOUR: H O U R;
+ID: I D;
+IF: I F;
+ILIKE: I L I K E;
+IN: I N;
+INDEX: I N D E X;
+INF: I N F | I N F I N I T Y;
+INJECTIVE: I N J E C T I V E;
+INNER: I N N E R;
+INSERT: I N S E R T;
+INTERVAL: I N T E R V A L;
+INTO: I N T O;
+IS: I S;
+IS_OBJECT_ID: I S UNDERSCORE O B J E C T UNDERSCORE I D;
+JOIN: J O I N;
+KEY: K E Y;
+KILL: K I L L;
+LAST: L A S T;
+LAYOUT: L A Y O U T;
+LEADING: L E A D I N G;
+LEFT: L E F T;
+LIFETIME: L I F E T I M E;
+LIKE: L I K E;
+LIMIT: L I M I T;
+LIVE: L I V E;
+LOCAL: L O C A L;
+LOGS: L O G S;
+MATERIALIZE: M A T E R I A L I Z E;
+MATERIALIZED: M A T E R I A L I Z E D;
+MAX: M A X;
+MERGES: M E R G E S;
+MIN: M I N;
+MINUTE: M I N U T E;
+MODIFY: M O D I F Y;
+MONTH: M O N T H;
+MOVE: M O V E;
+MUTATION: M U T A T I O N;
+NAN_SQL: N A N; // conflicts with macro NAN
+NO: N O;
+NOT: N O T;
+NULL_SQL: N U L L; // conflicts with macro NULL
+NULLS: N U L L S;
+OFFSET: O F F S E T;
+ON: O N;
+OPTIMIZE: O P T I M I Z E;
+OR: O R;
+ORDER: O R D E R;
+OUTER: O U T E R;
+OUTFILE: O U T F I L E;
+OVER: O V E R;
+PARTITION: P A R T I T I O N;
+POPULATE: P O P U L A T E;
+PRECEDING: P R E C E D I N G;
+PREWHERE: P R E W H E R E;
+PRIMARY: P R I M A R Y;
+PROJECTION: P R O J E C T I O N;
+QUARTER: Q U A R T E R;
+RANGE: R A N G E;
+RELOAD: R E L O A D;
+REMOVE: R E M O V E;
+RENAME: R E N A M E;
+REPLACE: R E P L A C E;
+REPLICA: R E P L I C A;
+REPLICATED: R E P L I C A T E D;
+RIGHT: R I G H T;
+ROLLUP: R O L L U P;
+ROW: R O W;
+ROWS: R O W S;
+SAMPLE: S A M P L E;
+SECOND: S E C O N D;
+SELECT: S E L E C T;
+SEMI: S E M I;
+SENDS: S E N D S;
+SET: S E T;
+SETTINGS: S E T T I N G S;
+SHOW: S H O W;
+SOURCE: S O U R C E;
+START: S T A R T;
+STOP: S T O P;
+SUBSTRING: S U B S T R I N G;
+SYNC: S Y N C;
+SYNTAX: S Y N T A X;
+SYSTEM: S Y S T E M;
+TABLE: T A B L E;
+TABLES: T A B L E S;
+TEMPORARY: T E M P O R A R Y;
+TEST: T E S T;
+THEN: T H E N;
+TIES: T I E S;
+TIMEOUT: T I M E O U T;
+TIMESTAMP: T I M E S T A M P;
+TO: T O;
+TOP: T O P;
+TOTALS: T O T A L S;
+TRAILING: T R A I L I N G;
+TRIM: T R I M;
+TRUNCATE: T R U N C A T E;
+TTL: T T L;
+TYPE: T Y P E;
+UNBOUNDED: U N B O U N D E D;
+UNION: U N I O N;
+UPDATE: U P D A T E;
+USE: U S E;
+USING: U S I N G;
+UUID: U U I D;
+VALUES: V A L U E S;
+VIEW: V I E W;
+VOLUME: V O L U M E;
+WATCH: W A T C H;
+WEEK: W E E K;
+WHEN: W H E N;
+WHERE: W H E R E;
+WINDOW: W I N D O W;
+WITH: W I T H;
+YEAR: Y E A R | Y Y Y Y;
+
+JSON_FALSE: 'false';
+JSON_TRUE: 'true';
+
+// Tokens
+
+// copied from clickhouse_driver/util/escape.py
+ESCAPE_CHAR
+    : BACKSLASH B
+    | BACKSLASH F
+    | BACKSLASH R
+    | BACKSLASH N
+    | BACKSLASH T
+    | BACKSLASH '0'
+    | BACKSLASH A
+    | BACKSLASH V
+    | BACKSLASH BACKSLASH
+    | BACKSLASH QUOTE_SINGLE;
+
```

Comment:
I added this part to the original CH grammar to be sure to reject strings that escape non-CH escape chars. So `\s` will fail.